cmd/apiserver/server_test.go

```diff
@@ -0,0 +1,462 @@
+					Spec:       calculationsv1.CalculationSpec{Teff: 11000.0, LogG: 4.0},
+				},
+				{
+					TypeMeta:   metav1.TypeMeta{Kind: "Calculation", APIVersion: "vega.io/v1"},
+					ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("calc-%s", util.InputHash([]byte(fmt.Sprintf("%f", 12000.0)), []byte(fmt.Sprintf("%f", 4.0))))},
+					Phase:      calculationsv1.CreatedPhase,
+					Status:     calculationsv1.CalculationStatus{StartTime: metav1.Time{Time: time.Date(2000, 0, 0, 0, 0, 0, 0, time.UTC)}},
+					Spec:       calculationsv1.CalculationSpec{Teff: 12000.0, LogG: 4.0},
+				},
+			},
+		},
+		{
+			id: "none out of X calculations get deleted",
+			initialCalculation: []calculationsv1.Calculation{
+				{
+					TypeMeta:   metav1.TypeMeta{Kind: "Calculation", APIVersion: "vega.io/v1"},
+					ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("calc-%s", util.InputHash([]byte(fmt.Sprintf("%f", 11000.0)), []byte(fmt.Sprintf("%f", 4.0))))},
+					Phase:      calculationsv1.CreatedPhase,
+					Status:     calculationsv1.CalculationStatus{StartTime: metav1.Time{Time: time.Date(2000, 0, 0, 0, 0, 0, 0, time.UTC)}},
+					Spec:       calculationsv1.CalculationSpec{Teff: 11000.0, LogG: 4.0},
+				},
+				{
+					TypeMeta:   metav1.TypeMeta{Kind: "Calculation", APIVersion: "vega.io/v1"},
+					ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("calc-%s", util.InputHash([]byte(fmt.Sprintf("%f", 12000.0)), []byte(fmt.Sprintf("%f", 4.0))))},
+					Phase:      calculationsv1.CreatedPhase,
+					Status:     calculationsv1.CalculationStatus{StartTime: metav1.Time{Time: time.Date(2000, 0, 0, 0, 0, 0, 0, time.UTC)}},
+					Spec:       calculationsv1.CalculationSpec{Teff: 12000.0, LogG: 4.0},
+				},
+			},
+			errorExpected: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		fakecs := fake.NewSimpleClientset()
+		fakeClient := fakecs.VegaV1()
+
+		s := server{
+			logger: logrus.WithField("test-name", tc.id),
+			ctx:    context.Background(),
+			client: fakeClient,
+		}
+
+		for _, calc := range tc.initialCalculation {
+			_, err := fakeClient.Calculations().Create(s.ctx, &calc, metav1.CreateOptions{})
+			if err != nil {
+				t.Errorf("Coudn't create the calculation: %v", calc)
+			} else {
+				logrus.WithFields(logrus.Fields{"name": calc.Name, "teff": calc.Spec.Teff, "logG": calc.Spec.LogG}).Info("Calculation successfully created")
+			}
+		}
+		req, err := http.NewRequest("DELETE", "/calculations/delete/", nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+		vars := make(map[string]string)
+		if !tc.errorExpected {
+			vars = map[string]string{
+				"id": tc.initialCalculation[0].Name,
+			}
+		} else {
+			vars = map[string]string{
+				"id": "calc-expect-error",
+			}
+		}
+		req = mux.SetURLVars(req, vars)
+
+		req.Header.Set("Content-Type", "application/json")
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(s.deleteCalculation)
+		handler.ServeHTTP(rr, req)
+
+		if status := rr.Code; status == http.StatusOK {
```

Comment:
this is ok but not sufficient. We should also get all the calculations and compare them with the expected calculations. Please add them in the testcase as well.